src/odemis/acq/align/autofocus.py

```diff
@@ -587,10 +591,258 @@ def estimateAutoFocusTime(detector, scanner=None, steps=MAX_STEPS_NUMBER):
+      for the "affecter"
+    affected (list of Component): set of affected components
+    return (Component): the first component that affects all the affected
+    raise LookupError: if no component found
+    """
+    naffected = set(c.name for c in affected)
+    for c in comps:
+        if c is not None and naffected <= set(c.affects.value):
+            return c
+    else:
+        raise LookupError("Failed to find a component that affects all %s" % (naffected,))
+
+
+def _DoSparc2AutoFocus(future, streams, align_mode, opm, dets, spgr, selector, focuser, start_autofocus):
+
+    try:
+
+        if future._autofocus_state == CANCELLED:
+            logging.warning("Autofocus procedure is cancelled before the light is on")
+            raise CancelledError()
+
+        logging.debug("Turn on the light")
+        # Turn on the light and check it is on
+        bl = model.getComponent(role="brightlight")
+        f = turnOnLightAndCheck(bl, dets[0])
+        try:
+            f.result(timeout=60)
+        except TimeoutError:
+            f.cancel()
+            logging.warning("Light doesn't appear to turn on after 60s")
+        if future._autofocus_state == CANCELLED:
+            logging.warning("Autofocus procedure is cancelled after turning on the light")
+            raise CancelledError()
+        future._actions_time.pop(0)
+        future.set_progress(end=time.time() + sum(future._actions_time))
+
+        logging.debug("Adjust the optical path")
+        # Go to the special focus mode
+        if align_mode == "lens-align":
+            opath = "spec-focus"
+        elif align_mode == "fiber-align":
+            opath = "spec-fiber-focus"
+        # change the optical path
+        fopm = opm.setPath(opath)
+        fopm.result()
+        if future._autofocus_state == CANCELLED:
+            logging.warning("Autofocus procedure is cancelled after closing the slit")
+            raise CancelledError()
+        future._actions_time.pop(0)
+        future.set_progress(end=time.time() + sum(future._actions_time))
+
+        # Autofocus is not manual
+        if start_autofocus == True:
+            # Configure each detector with good settings
+            for d in dets:
+                # The stream takes care of configuring its detector, so no need
+                if model.hasVA(d, "binning"):
+                    d.binning.value = d.binning.clip((2, 2))
+                if model.hasVA(d, "exposureTime"):
+                    d.exposureTime.value = d.exposureTime.clip(0.2)
+            ret = {}
+            logging.debug("Run AutoFocusSpectrometer")
+            try:
+                # run AutoFocusSpectrometer
+                logging.debug("AutoFocusSpectrometer starts")
+                try:
+                    future._running_subf = AutoFocusSpectrometer(spgr, focuser, dets, selector, streams)
+                    ret = future._running_subf.result(timeout=100)
```

Comment:
You need to increase the auto focus timeout (it can take a long time). Make it 500s.